test/quantization/test_auto_tracing.py

```diff
@@ -0,0 +1,924 @@
+import collections
+import copy
+import math
+import unittest
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.nn.intrinsic as nni
+from torch.testing._internal.common_quantization import (
+    skipIfNoFBGEMM,
+    skip_if_no_torchvision,
+    QuantizationTestCase,
+)
+from torch.quantization import (
+    ObserverBase,
+    FakeQuantizeBase,
+)
+from torch.quantization.quantize_fx import (
+    prepare_fx,
+    convert_fx,
+)
+
+import torch.ao.quantization._quantize_dynamic_tracing as _quantize_dynamic_tracing
+
+def _allclose(a, b):
+    if isinstance(a, tuple):
+        assert isinstance(b, tuple)
+        result = True
+        for a_inner, b_inner in zip(a, b):
+            result = result and torch.allclose(a_inner, b_inner)
+        return result
+    elif isinstance(a, torch.Tensor):
+        assert isinstance(b, torch.Tensor)
+        return torch.allclose(a, b)
+    raise AssertionError('unhandled type')
+
+
+class AutoTracingTestCase(QuantizationTestCase):
+    def _test_auto_tracing(
+        self,
+        m,
+        qconfig,
+        example_args,
+        fuse_modules=True,
+        do_fx_comparison=True,
+        do_torchscript_checks=True,
+    ):
+        m_copy = copy.deepcopy(m)
+
+        m.qconfig = qconfig
+
+        mp = _quantize_dynamic_tracing.prepare(
+            m, example_args, fuse_modules=fuse_modules)
+        out_p = mp(*example_args)
+        # print(mp)
```

Comment:
Can we remove all the commented out prints here?

Reply:
would it be ok if we did this in a couple of weeks? This is still early stage and these are useful for debugging.